data_structures/src/chain.rs

```diff
@@ -1798,21 +1806,28 @@ impl TransactionsPool {
     /// # use witnet_data_structures::transaction::{Transaction, VTTransaction};
     /// let mut pool = TransactionsPool::new();
     /// let transaction = Transaction::ValueTransfer(VTTransaction::default());
-    /// pool.insert(transaction);
+    /// pool.insert(transaction, 0);
     ///
     /// assert!(!pool.is_empty());
     /// ```
-    pub fn insert(&mut self, transaction: Transaction) {
-        let weight = 0; // TODO: weight = transaction-fee / transaction-weight
+    #[allow(clippy::cast_precision_loss)]
+    pub fn insert(&mut self, transaction: Transaction, fee: u64) {
         let key = transaction.hash();
 
         match transaction {
             Transaction::ValueTransfer(vt_tx) => {
-                self.vt_transactions.insert(key, (weight, vt_tx));
-                self.sorted_index.insert((weight, key));
+                let weight = std::cmp::max(u64::from(vt_tx.weight()), 1) as f64;
```

Comment:
I think that if you remove this `u64::from` you can also remove `     #[allow(clippy::cast_precision_loss)]`
```suggestion
                let weight = std::cmp::max(vt_tx.weight(), 1) as f64;
```

Reply:
Also, dividing by 0 is not a problem with floats, `1.0/0.0` is `+inf`, so a transaction with 0 weight will have maximum priority.

Reply:
_Infinite priority_